shell/platform/windows/platform_handler_win32.cc

```diff
@@ -149,40 +153,44 @@ bool ScopedClipboard::HasString() {
          ::IsClipboardFormatAvailable(CF_TEXT);
 }
 
-std::optional<std::wstring> ScopedClipboard::GetString() {
+std::variant<std::wstring, int> ScopedClipboard::GetString() {
   assert(opened_);
 
   HANDLE data = ::GetClipboardData(CF_UNICODETEXT);
   if (data == nullptr) {
-    return std::nullopt;
+    return ::GetLastError();
   }
   ScopedGlobalLock locked_data(data);
   if (!locked_data.get()) {
-    return std::nullopt;
+    return ::GetLastError();
+  }
+  std::optional<std::wstring> string = static_cast<wchar_t*>(locked_data.get());
+  if (!string) {
+    return ::GetLastError();
```

Comment:
What is the purpose of this new code? I'm not sure why there's an optional, or why you're checking for failure since I'm not clear on how the string creation would fail.

Reply:
I think the way I wrote this was redundant.  I've updated it so it will return an error if locked_data.get fails, otherwise it returns the string.

Reply:
I'm still not sure why there's a `std::optional` at all. Why not just:

```
if (!locked_data.get()) {
  return ::GetLastError();
}
return std::string(static_cast<wchar_t*>(locked_data.get()));
```